src/videojs-contrib-hls.js

```diff
@@ -87,6 +87,9 @@ export default class HlsHandler extends Component {
     // HlsSourceHandler.handleSource()
     // @link https://github.com/videojs/video.js/pull/3245
     this.options_ = videojs.mergeOptions(options, videojs.options.hls || {});
+    // we cannot override the mode at this state, we have to use what was passed
+    // to us
+    this.options_.mode = options.mode;
```

Comment:
It might be a good thing to throw a helpful error in that case. It seems to me this would only occur because of programmer misconfiguration and normally it's good to point that out during development.